src/Microsoft.IdentityModel.Tokens/Validation/Results/ValidationResult.cs

```diff
@@ -58,25 +59,60 @@ public ValidationResult(ValidationError error)
         public static implicit operator ValidationResult<TResult>(ValidationError error) => new(error);
 
         /// <summary>
-        /// Gets a value indicating whether the result is successful.
+        /// Gets a value indicating whether the result is valid.
         /// </summary>
-        public readonly bool IsSuccess { get; }
+        public readonly bool IsValid { get; }
 
         /// <summary>
         /// Unwraps the result.
         /// </summary>
         /// <returns>The wrapped result value.</returns>
-        /// <remarks>This method is only valid if the result type is successful.</remarks>
-        /// <exception cref="InvalidOperationException">Thrown if attempted to unwrap the value from a failed result.</exception>
-        public TResult UnwrapResult() => IsSuccess ? _result! : throw new InvalidOperationException("Cannot unwrap error result");
+        /// <remarks>This method is only valid if the result type is valid.</remarks>
+        /// <exception cref="InvalidOperationException">Thrown if attempted to unwrap the value from a non valid result.</exception>
+        internal TResult UnwrapResult() => IsValid ? _result! : throw new InvalidOperationException("Cannot unwrap error result");
 
         /// <summary>
         /// Unwraps the error.
         /// </summary>
         /// <returns>The wrapped error value.</returns>
-        /// <remarks>This method is only valid if the result type is unsuccessful.</remarks>
-        /// <exception cref="InvalidOperationException">Thrown if attempted to unwrap an error from a successful result.</exception>
-        public ValidationError UnwrapError() => IsSuccess ? throw new InvalidOperationException("Cannot unwrap success result") : _error!;
+        /// <remarks>This method is only valid if the result type is not valid.</remarks>
+        /// <exception cref="InvalidOperationException">Thrown if attempted to unwrap an error from a valid result.</exception>
+        internal ValidationError UnwrapError() => IsValid ? throw new InvalidOperationException("Cannot unwrap success result") : _error!;
+
+        /// <summary>
+        /// Gets the error associated with the validation result.
+        /// </summary>
+        /// <returns>The error associated with the validation result.</returns>
+        /// <remarks>This property is only valid if the result type is not valid.</remarks>
+        public ValidationError? Error
+        {
+            get
+            {
+                if (IsValid)
+                    LogHelper.LogWarning("Warning: Accessing the Error property in a valid result is invalid.");
```

Comment:
nit: I wonder if we should have this as an IDX message?

Reply:
I'll raise an issue to review the messages and look into that as well.